### DISCUSSION TITLE
Can (magit-log-other) opportunistically default to thing at point?

### DISCUSSION THREAD

**Discussion by brc** [magit/magit#5205] [Ideas]
# First...
Thank you for your tireless work and support, @tarsius. (I am a big fan of your work, and I'm a sponsor!)

I am very happily using:
```
Magit 20240821.158 [>= 4.0.0], Transient 0.7.4, Git 2.44.0, Emacs 29.2, gnu/linux
```
# TL;DR Question

What is the main reason that `(magit-log-other)` (using `l o`) does not default to the ref at point?

# My Guess / Observations

The docstring for this function implies it may be by design that we require user input due to the fact the user may want to provide free-form input to this command (such as a range) which cannot be known in advance, making it impossible to open a list of candidates by default.

AFAICT by browsing the Magit source code for the first time, however, `(magit-log-read-revs)` _does_ call `(magit-branch-or-commit-at-point)` to obtain a value for the `def` arg of `(magit-completing-read-multiple)` (that is: the seventh argument, which eventually becomes the `DEFAULT-VALUE` argument to `(read-from-minibuffer)`). But, in my years of using Magit, whenever I press `l o` in the `magit-refs` buffer while point is located on a ref name, I only ever get the empty minibuffer prompt `"Log rev,s: "` and I must start typing something to eventually complete the candidate name. Is this working as intended? I manually evaluated `(magit-branch-or-commit-at-point)` in the `magit-refs` buffer to make sure it worked and it of course returned the correct value.

# Use Case

As we all agree, one of the very best parts about the Magit UI is dynamically interacting with objects. When browsing the `magit-refs` buffer (with `y r`), I may conveniently invoke `(magit-checkout)` using `b b` or `(magit-branch-delete)` using `b x`, and the ref at point becomes the default argument so that I may simply strike `RET` (with authority, on a mechanical keyboard, as if I know what I'm doing). In this situation, I would much prefer the default behavior of `l o` to be the same way; in the lesser chance that I wish to type something that cannot be discovered in advance (e.g., range), I would prefer _that_ workflow to require extra keystrokes&#8212;not the other way around.  Does anyone else agree?

Thank you, Jonas and community!

**Comment by tarsius** [comment]
That ref was actually already used as the default. The problem was that the default was not shown in the prompt, so you weren't aware of that (and also couldn't be sure what would happen when you press `RET` without any input).

I wasn't aware of this because I use a completion framework for which it would be redundant to do that. (A list of completion candidates is instantly displayed and if there is a default, then it is pre-selected and we can now that because it is highlighted.)

`magit-completing-read` did already automatically append the default to the prompt, but only if appropriate. `magit-completing-read-multiple` however, never appended the default. I am surprised it took this long for someone to report this, as it's obviously quite annoying.

Fixed in 461d5ba249b8178ae83f80596da4312e65010720.

**Reply by brc** [reply]
Awesome. Glad I'm not misunderstanding anything too badly. Thank you for the response and very fast fix!! 👏 

> I am surprised it took this long for someone to report this

Magit is so polished at this point that I tend to assume the problem lies with me and not Magit. But, I wonder how often people use the workflow I am describing (`y r` then `l o`). Sounds like completion framework may be an important factor.
Unfortunately, I work in a group that creates many dozens of branches all named by Jira issues (eg., `MAG-125497`). So I am [more frequently than I'd like to be] inspecting these branches to look at commit logs.

> it's obviously quite annoying

It's just exacerbated in my current situation because: Imagine pages of branch names that I described above&#8212;visually very painful to look at but also not very useful to fuzzy-find any of them by name. I've been dealing with this workflow regularly for the last 2.5 years, so it's cumbersome when point is already located on a branch name I want to inspect, but I must start typing `MAG` then `<TAB>` (to open Helm, in my case), and then type all six numbers. :-) `</vent>`

I'll go read docs and figure out how to use the latest tree. Thanks again.
**EDIT:** ...and I'll just try pressing `RET` at the prompt until then.
